### ISSUE TITLE
Update collection code for editing Mozilla collections

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#7325]
The code to edit a Mozilla collection used to rely on the (`mozilla`) username. We now use the `userId` so the code should be updated for that.

**For QA:** please make sure a user who is allowed to edit a mozilla collection can edit Mozilla collections. See also: #5163.

**Comment by ioanarusiczki** [comment]
@willdurand 
I verified this issue on AMO dev - FF64(Win10). 
https://github.com/mozilla/addons-server/issues/8509 was a good guide for testing if adding or removing add-ons or themes from a Mozilla collection can be done with different type of users. 
I tested with a collection transferred by a user to Mozilla, Password Managers, Weather reports, News boosters and Featured themes.

Here are my results:
-  `Admin` users can add or remove extensions using django admin and the new front end.
-  `Collections:Contribute` can add or remove themes for the Featured Theme collection from django admin. Right now the issue https://github.com/mozilla/addons-server/issues/10151 that mentions the link that was available to edit the collection from the frontend is no longer available https://addons-dev.allizom.org/en-US/firefox/collections/mozilla/featured-personas/. The link returns a server error message. This is reproducible with Admin accounts and content curators too.
 Also, I filed an issue on server, I could only add themes after clicking "Save" in django admin twice https://github.com/mozilla/addons-server/issues/10403
- `Admin:Curation` can add or remove extensions using django admin and the new front end.
- users with `Collections:Edit` cannot edit these collections using the frontend. A user from `Staff` or `Admin advanced` can remove extensions from collections using django admin but they cannot add any (there is no button available). A user who belongs to `Admin advanced` can also delete a collection using django admin (using the `Delete` red button). 

![advanced admin group](https://user-images.githubusercontent.com/33448286/51185461-db728580-18df-11e9-856e-52f72db96a5c.png)

**Comment by willdurand** [comment]
@ioanarusiczki so, it looks like it "works" just as before on the new frontend? (the django admin is out of scope here)

**Comment by ioanarusiczki** [comment]
@willdurand On the front end I think it works as before. 
The only problem which I filed it a while ago on server is that the frontend link for editing featured themes is not working anymore. I updated the server issue https://github.com/mozilla/addons-server/issues/10151 but maybe I should transfer it on the frontend repo.
The link is https://addons-dev.allizom.org/en-US/firefox/collections/mozilla/featured-personas/
and won't work (tried it with all user permissions)

![server error](https://user-images.githubusercontent.com/33448286/51318106-0a186980-1a62-11e9-8a18-e5ffce34cd29.png)
